### DISCUSSION TITLE
CURL_DISABLE_IPFS anyone?

### DISCUSSION THREAD

**Discussion by gvanem** [curl/curl#12935] [Q&A]
Grepping thought the sources, I found 41 unique instances of `CURL_DISABLE*`. How many are there really?
But not any `CURL_DISABLE_IPFS`.  I bet someone would consider IPFS harmful (facilitate a lite version of the *dark web* 
with illegal content etc). 

Since there is a `CURL_DISABLE_PROXY` to stop curl using a *Tor socks5* proxy, there should perhaps be a
 `CURL_DISABLE_IPFS` to stop the use of a *IPFS-gateway*? I failed to see there is this option today.

This is the full list I found:
```c
CURL_DISABLE_ALTSVC
CURL_DISABLE_AWS
CURL_DISABLE_BASIC_AUTH
CURL_DISABLE_BEARER_AUTH
CURL_DISABLE_BINDLOCAL
CURL_DISABLE_COOKIES
CURL_DISABLE_CRYPTO_AUTH
CURL_DISABLE_DEPRECATION
CURL_DISABLE_DICT
CURL_DISABLE_DIGEST_AUTH
CURL_DISABLE_DOH
CURL_DISABLE_FILE
CURL_DISABLE_FORM_API
CURL_DISABLE_FTP
CURL_DISABLE_GOPHER
CURL_DISABLE_HEADERS_API
CURL_DISABLE_HSTS
CURL_DISABLE_HTTP
CURL_DISABLE_HTTP_AUTH
CURL_DISABLE_IMAP
CURL_DISABLE_KERBEROS_AUTH
CURL_DISABLE_LDAP
CURL_DISABLE_LDAPS
CURL_DISABLE_LIBCURL_OPTION
CURL_DISABLE_MIME
CURL_DISABLE_MQTT
CURL_DISABLE_NEGOTIATE_AUTH
CURL_DISABLE_NETRC
CURL_DISABLE_PARSEDATE
CURL_DISABLE_POP3
CURL_DISABLE_PROGRESS_METER
CURL_DISABLE_PROXY
CURL_DISABLE_RTSP
CURL_DISABLE_SHUFFLE_DNS
CURL_DISABLE_SMB
CURL_DISABLE_SMTP
CURL_DISABLE_SOCKETPAIR
CURL_DISABLE_TELNET
CURL_DISABLE_TFTP
CURL_DISABLE_TYPECHECK
CURL_DISABLE_VERBOSE_STRINGS
```

**Comment by bagder** [comment]
They are documented here: https://github.com/curl/curl/blob/master/docs/CURL-DISABLE.md

**Comment by gvanem** [comment]
> They are documented here: 

Ok. Nice to know.
